javaparser-core/src/main/javacc/java.jj

```diff
@@ -2101,7 +2112,19 @@ SimpleName SimpleName():
     SimpleName ret;
 }
 {
-  <IDENTIFIER> { ret = new SimpleName(tokenRange(), token.image); }
+  Identifier() { ret = new SimpleName(tokenRange(), token.image); }
+  { return ret; }
+}
+
+
+String Identifier():
+{
+    String ret;
+}
+{
+    // Make sure the module info keywords don't interfere with normal Java parsing by matching them as normal identifiers.
+  (<MODULE> | <REQUIRES> | <TO> | <WITH> | <OPEN> | <OPENS> | <USES> | <EXPORTS> | <PROVIDES> | <TRANSITIVE> |
```

Comment:
I would:
* parse something that has a name
* change it to a name corresponding to a module keyword
* print it using lexical preservation

Do the viceversa (starting from something having a name like "requires").

These tests would be very similar to the ones in the lexicalpreservation package